[Draft] [docs] Add Backend to directory structure (#4503)
p4lang/p4c: docs/README.md

```
p4c
├── build                     -- recommended place to build binary
├── backends          

> │   ├── bmv2                  -- behavioral model version 2 (switch simulator) back-end

Reviewer: ```suggestion
│   ├── bmv2                  -- behavioral model version 2 (switch simulator) back end
```

> │   ├── common                -- common back-end code

Reviewer: ```suggestion
│   ├── common                -- code shared by multiple back ends
```

> │   ├── dpdk                  -- translates the P4-16 programs to DPDK API

Reviewer: ```suggestion
│   ├── dpdk                  -- translates P4 code to DPDK instructions
```

> │   ├── ebpf                  -- extended Berkeley Packet Filters back-end

Reviewer: ```suggestion
│   ├── ebpf                  -- extended Berkeley Packet Filters back end
```

│   ├── graphs                -- backend that can draw graphiz graphs of P4 programs

> │   ├── p4test                -- "fake" back-end for testing

Reviewer: ```suggestion
│   ├── p4test                -- "fake" back end for testing
```

> │   ├── p4test                -- "fake" back-end for testing
> │   ├── p4tools               -- shared C++ components for the various P4Tools modules
> │   ├── tc                    -- generates files for the P4TC implementation
> │   └── ubpf                  -- translate P4 programs into the uBPF programs

Reviewer: ```suggestion
│   ├── p4test                -- "fake" back-end for testing
│   ├── p4tools               -- testing library for different static analysis tools
│   ├── tc                    -- translates P4 code to Linux TC rules
│   └── ubpf                  -- translates P4 programs to uBPF code
```
